src/x86/sse2.rs

```diff
@@ -392,6 +428,25 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_mm_storel_epi64_cell() {
+        unsafe { test() }
+
+        #[target_feature(enable = "sse2")]
+        fn test() {
+            let mut a = [0, 1, 2, 3, 4];
+            let val = core::cell::Cell::from_mut(&mut a[..]).as_slice_of_cells();
+
+            let load: &[_; 4] = val[..4].try_into().unwrap();
+            let store: &[_; 4] = val[1..].try_into().unwrap();
+
+            let r = super::_mm_loadu_si128_cell(load);
+            super::_mm_storeu_si128_cell(store, r);
+
+            assert_eq!(a, [0, 0, 1, 2, 3]);
+        }
+    }
```

Comment:
This isn't exercising `_mm_loadl_epi64_cell` and `_mm_storel_epi64_cell`.

Reply:
The intrinsics guide and stdarch both have pointers to `__m128i` 🤷 

https://doc.rust-lang.org/1.88.0/core/arch/x86/fn._mm_loadl_epi64.html
```rust
pub unsafe fn _mm_loadl_epi64(mem_addr: *const __m128i) -> __m128i
```

https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#ssetechs=SSE2&text=loadl_epi64&ig_expand=4044
```c++
__m128i _mm_loadl_epi64 (__m128i const* mem_addr)
```

Reply:
It's supposed to emit the same `movq` instruction as `_mm_loadu_si64` which takes a void pointer (I implemented it as `&[u8; 8]` but it could be its own `Is64BitUnaligned` trait).
```cpp
__m128i _mm_loadu_si64 (void const* mem_addr)
```

There are a few weird/inconsistent signatures like this one. It doesn't have to be aligned and should only need 64-bit but is 128-bit for some reason. 

For implementing the intrinsics, I went through the intrinsics guide with the Load and Store categories enabled and proceeded through the instruction families from SSE to AVX2 for completeness while comparing to the arch docs. There are all sorts of oddities like inconsistent signatures, inconsistent naming, and duplicates.

Reply:
Alright. Since there's no harm in requiring more data to be behind the pointer, in particular with the Cell allowing overlaps I'll keep the implementation consistent with the current non-cell versions, test them and then let you decide on fixing all of them with a new trait or not.